File: app/src/main/java/com/kkkkkn/readbooks/view/customView/LoadingDialog.java
package com.kkkkkn.readbooks.view.customView;

import android.app.Dialog;
import android.content.Context;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import com.kkkkkn.readbooks.R;

public class LoadingDialog extends Dialog {
    private AppCompatTextView tv_msg;

    public LoadingDialog(@NonNull Context context) {
        super(context);
        setContentView(R.layout.loading_dialog);
        setCanceledOnTouchOutside(false);
        tv_msg = findViewById(R.id.loading_msg);
    }

    public void showLoading(String content) {
        super.show();
        tv_msg.setText(content);
    }
    public void hideLoading(){
        super.dismiss();
    }

}

File: app/src/main/java/com/kkkkkn/readbooks/view/customView/LoadingDialog.kt
package com.kkkkkn.readbooks.view.customView

import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import com.kkkkkn.readbooks.databinding.LoadingDialogBinding

class LoadingDialog(context: Context) : Dialog(context) {
    private var viewBinding: LoadingDialogBinding

    init {
        viewBinding= LoadingDialogBinding.inflate( LayoutInflater.from(context),null,false )
        this.setContentView(viewBinding.root)
        setCanceledOnTouchOutside(false)
    }

    fun showLoading(content: String?) {
        super.show()
        viewBinding.loadingMsg.text = content
    }

    fun hideLoading() {
        super.dismiss()
    }
}